src/tests/testHtmlQuote.cc

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+#include "compat/cppunit.h"
+#include "html_quote.h"
+#include "unitTestMain.h"
+
+#include <cstring>
+#include <iostream>
+
+class testHtmlQuote: public CPPUNIT_NS::TestFixture
+{
+    CPPUNIT_TEST_SUITE(testHtmlQuote);
+    CPPUNIT_TEST(test_html_quote_cstr);
+    // CPPUNIT_TEST(testPerformance);
+    CPPUNIT_TEST_SUITE_END();
+
+    protected:
+    void test_html_quote_cstr();
+    void testPerformance();
+};
+
+CPPUNIT_TEST_SUITE_REGISTRATION( testHtmlQuote );
+
+void
+testHtmlQuote::test_html_quote_cstr()
+{
+    CPPUNIT_ASSERT_EQUAL(std::string(""), std::string(html_quote("")));
+    CPPUNIT_ASSERT_EQUAL(std::string("bar"), std::string(html_quote("bar")));
+    CPPUNIT_ASSERT_EQUAL(std::string("foo&lt;bar&gt;gazonk"), std::string(html_quote("foo<bar>gazonk")));
+    CPPUNIT_ASSERT_EQUAL(std::string("foo&amp;bar"), std::string(html_quote("foo&bar")));
+    CPPUNIT_ASSERT_EQUAL(std::string("some&#39;thing"), std::string(html_quote("some'thing")));
+    CPPUNIT_ASSERT_EQUAL(std::string("some&quot;thing"), std::string(html_quote("some\"thing")));
+    CPPUNIT_ASSERT_EQUAL(std::string("&#31;"), std::string(html_quote("\x1f")));
+}
```

Comment:
Please add the following test cases:

* one test for every HTML entity (see htmlEntities)
* a single test with a string that has nothing but a single HTML entity (of your choice)
* a single test with a string that has nothing but a code point (of your choice) that needs `&#nnn` encoding

Reply:
added